### DISCUSSION TITLE
Triangular dependencies in libraries.

### DISCUSSION THREAD

**Discussion by markfoodyburton** [cpm-cmake/CPM.cmake#226] [Q&A]
I have library A which is "well behaved" and uses export to make a library.
Library B likewise, and uses a find_package(A) to find A.
Then I have a project C, which uses CPMFindPackage to include A and likewise B.

I then get:

CMake Error at ..../_deps/A/AConfig.cmake: (include):
  The file
    .../ATargets.cmake

  was generated by the export() command.  It may not be used as the argument
  to the include() command.  Use ALIAS targets instead to refer to targets by
  alternative names.

Call Stack (most recent call first):
  .../B/CMakeLists.txt (find_package)

Building B on it's own works... so I assume It's an issue somehow with CPM, but I dont understand how !

Any help appreciated !

**Comment by TheLartians** [comment]
Hm my personal projects have a lot of triangular dependencies, however they all use `CPMAddPackage` for managing dependencies. Perhaps the issue arises from inside the the `AConfig.cmake` or a conflict with the generated `findA.cmake` modules. It's hard to say without a reproducible example.

**Comment by markfoodyburton** [comment]
Looks like you were right.... thanks for the pointer
